### PR TITLE
Removed sharedassembyingfo and replaced with updating project.json

### PR DESCRIPTION
**Pull request by jchannon**
This sets the Nancy project.json.

**NOTE** : Do we need to set all project.json versions? @khellang @thecodejunkie?

### CODE REVIEW

**Comment by jchannon** [comment]
**UPDATE** Changed it to loop over all project.json files

File: build.sh [hunk 1]
```diff
@@ -30,8 +31,8 @@ function installdotnet() {
     then
       echo "Installing dotnet"
       apt-get install curl
-      curl -sSL https://raw.githubusercontent.com/dotnet/cli/rel/1.0.0-preview1/scripts/obtain/dotnet-install.sh | bash /dev/stdin --version 1.0.0-preview1-002702 --install-dir ~/dotnet
-      ln -s ~/dotnet/dotnet /usr/local/bin
+      curl -sSL https://raw.githubusercontent.com/dotnet/cli/rel/1.0.0/scripts/obtain/dotnet-install.sh | bash /dev/stdin --channel beta --version 1.0.0-preview1-002702 --install-dir .dotnet
+      ln -s .dotnet/dotnet /usr/local/bin
```
Comment (devlead) [inline build.sh]: This shouldn't be needed right? should just be able to prepend to path?
Like the windows version:
https://github.com/patriksvensson/Nancy/blob/cake/build.ps1#L42-L45

File: build.cake [hunk 2]
```diff
@@ -170,85 +162,70 @@ Task("Publish-NuGet")
...
-      Description = info.Description,
-      InformationalVersion = informationalVersion ?? info.AssemblyInformationalVersion,
-      Product = info.Product,
-      Title = info.Title,
-      Version = assemblyVersion ?? info.AssemblyVersion
-    });
+  Verbose(logAction => logAction("Setting version to {0}", version));
+  foreach (var file in filePaths) 
+  {
+    var projectJsonModel = DeserializeJsonFromFile<dynamic>(file.FullPath);
```
Comment (devlead) [inline build.cake]: Dynamic won't work with the current version of Roslyn that's used.

File: build.sh (hunk 1 above)
Comment (jchannon) [inline build.sh]: `ln` will add it to PATH

File: build.cake (hunk 2 above)
Comment (jchannon) [inline build.cake]: Works on Mono or is that different?

File: build.sh (hunk 1 above)
Comment (devlead) [inline build.sh]: `ln` add an link from one file path to another.

`export PATH=PATH_TO_DOTNET:$PATH` would add it to path.

Comment (jchannon) [inline build.sh]: Sorry yeah

Comment (jchannon) [inline build.sh]: However because its in usr local bin it should be available to all suers

File: build.cake (hunk 2 above)
Comment (devlead) [inline build.cake]: Code is correct, it's a issue with the Roslyn version we're using.

But you could replace `#addin "Cake.Json"` with `#addin "Newtonsoft.Json"`
And replace `UpdateProjectJsonVersion` method with something like
```cake
public void UpdateProjectJsonVersion(string version, IEnumerable<FilePath> filePaths)
{
    Verbose("Setting version to {0}", version);
    foreach (var file in filePaths)
    {
        var project = Newtonsoft.Json.Linq.JObject.Parse(
            System.IO.File.ReadAllText(file.FullPath, Encoding.UTF8));
        project["version"].Replace(version);
        System.IO.File.WriteAllText(file.FullPath, project.ToString(), Encoding.UTF8);
    }
}
```

Comment (devlead) [inline build.cake]: `[Tt]ools/Addins/` should be added to `.gitignore`

File: build.sh (hunk 1 above)
Comment (devlead) [inline build.sh]: True, but isn't this semantically different vs. how build.ps1 works?

It'll look for global if not found it'll install locally to the repo,

Comment (jchannon) [inline build.sh]: Thats what build.ps1 does now

**Pull request merged**